[Android] Fix shadows size in clipped views

jsuarezruiz opened the pull request:
### Description of Change

Fix shadows size in clipped views on Android.

![Screenshot_1669222539](https://user-images.githubusercontent.com/6755973/203751507-5207ab08-e06f-4cd0-ac46-cb376354c0c0.png)

To validate the changes use the .NET MAUI Gallery and navigate to the Shadows page. If the shadow from the clipped view have the correct size, changes are validated.

![image](https://user-images.githubusercontent.com/6755973/218061563-b03ecad8-4161-4a1d-b740-cb12c0220175.png)

### Issues Fixed

Fixes #11578

jsuarezruiz committed:
Fix shadows clipping views

rmarinho reviewed (changes requested):
Test is failing 
<img width="436" alt="image" src="https://user-images.githubusercontent.com/1235097/203778743-8ea0e5b6-8b52-475b-8add-e391ab42011f.png">

jsuarezruiz committed:
Fix build errors

jsuarezruiz commented:
> Test is failing
> <img alt="image" width="436" src="https://user-images.githubusercontent.com/1235097/203778743-8ea0e5b6-8b52-475b-8add-e391ab42011f.png">

Leave a small change behind, fixed.

Redth committed:
Capture and cache Context usage

We've found Context can be a bit expensive to get in a large frequency of calls (https://github.com/dotnet/maui/pull/8001), so we can avoid this by capturing and caching the instance.

Redth commented:
@jsuarezruiz I pushed one small change to capture the Context and cache its use in a couple places in this view.

jsuarezruiz commented:
> @jsuarezruiz I pushed one small change to capture the Context and cache its use in a couple places in this view.

Better, thanks.

jsuarezruiz commented:
> > Test is failing
> > <img alt="image" width="436" src="https://user-images.githubusercontent.com/1235097/203778743-8ea0e5b6-8b52-475b-8add-e391ab42011f.png">
> 
> Leave a small change behind, fixed.

@rmarinho Could you review it again?. Thanks.

rachelkang committed:
Merge branch 'main' into fix-11578

rachelkang reviewed (approved):
LGTM! Ready to go once all tests are passing

jsuarezruiz committed:
Removed duplicated test

jsuarezruiz commented on src/Core/src/Platform/Android/WrapperView.cs:
```diff
@@ -268,7 +270,8 @@ void DrawShadow(Canvas canvas)
 					{
 						var bounds = new Graphics.RectF(0, 0, canvas.Width, canvas.Height);
-						var path = Clip.PathForBounds(bounds)?.AsAndroidPath();
+						var density = _context.GetDisplayDensity();
```
@Redth Thoughts?

jsuarezruiz replied:
Rui, reviewed and applied changes. Could you review it again?

jsuarezruiz committed:
Merge branch 'main' into fix-11578

jsuarezruiz committed:
Merge branch 'main' into fix-11578

jsuarezruiz committed:
Revert latest changes
